### PR TITLE
Updating FilterService to no longer crash on start on Android O

### PR DESCRIPTION
**Pull request by automaton82**
Right now if you start red-moon on Android O **without** the draw on top permission, the app will crash after 5 seconds. The cause of this is explained here:

https://stackoverflow.com/questions/44425584/context-startforegroundservice-did-not-then-call-service-startforeground

Essentially:

"The system allows apps to call Context.startForegroundService() even while the app is in the background. However, the app must call that service's startForeground() method within five seconds after the service is created."

Android O behaviour change -- https://developer.android.com/about/versions/oreo/android-8.0-changes.html

### CODE REVIEW

**Review by smichel17 — APPROVED** [review]
Sorry, not sure how I missed the notification for this.

Since apparently I am not as attentive here as I might be, I've changed the permissions here a little bit.

- Pushing directly to `master` is not allowed; all code must be added via PR
- PRs require 1 approval in order to merge them
- Anybody with the `maintainer` role can merge a PR
- Creating a tag that will trigger an f-droid release is also limited to maintainers only.

The current maintainers are: @AdamNiederer, @david-hil, @notjuliee, @raatmarien, and me. …and now you (if you accept the invitation)!

tl;dr maintainers are able to do everything needed to update & release the app now, but they have to do it in a pretty audit-able way; this allows me to give out the maintainer role pretty freely to anybody who shows an interest in helping keep the app up to date.
